src/haproxy/address.go

```diff
@@ -0,0 +1,66 @@
+package haproxy
+
+import (
+	"fmt"
+	"net"
+
+	proxyproto "github.com/pires/go-proxyproto"
+)
+
+// NewAddr creates an `Addr` with supplied network, host and port as strings.
+func NewAddr(network, host, port string) Addr {
+	return Addr{
+		net:  network,
+		addr: net.JoinHostPort(host, port),
+	}
+}
+
+// NewProxyAddr creates an `Addr` struct from supplied `proxyproto.AddressFamilyAndProtocol`, `net.IP` and port as `unit16`.
+func NewProxyAddr(proto proxyproto.AddressFamilyAndProtocol, addr net.IP, port uint16) Addr {
+	return Addr{
+		net:  convertProxyProtocolToString(proto),
+		addr: net.JoinHostPort(addr.String(), fmt.Sprintf("%d", port)),
+	}
+}
+
+// Addr is a `net.Addr` compatible struct for use with `net.Conn` `RemoteAddr()` and `LocalAddr()`
+type Addr struct {
```

Comment:
Is there any particular reason to use a custom `net.Addr` implementation here? The standard library already provides `TCPAddr` and `UnixAddr`.  These could be constructed based on the information available in the `AddressFamilyAndProtocol` value.

Reply:
Because I didn't see them when I was looking for `net.Addr` implementations :)

Reply:
Removed and replaced with `net.UDPAddr`, `net.TCPAddr` and `net.UnixAddr`